apps/emqx_redis/src/emqx_redis.erl

```diff
@@ -242,24 +243,57 @@ on_get_status(_InstId, #{type := cluster, pool_name := PoolName}) ->
                 [] ->
                     disconnected;
                 [_ | _] ->
-                    Health = eredis_cluster:ping_all(PoolName),
-                    status_result(Health)
+                    do_cluster_status_check(PoolName, State)
             end;
         false ->
             disconnected
     end;
-on_get_status(_InstId, #{pool_name := PoolName}) ->
-    Health = emqx_resource_pool:health_check_workers(PoolName, fun ?MODULE:do_get_status/1),
-    status_result(Health).
+on_get_status(_InstId, #{pool_name := PoolName} = State) ->
+    Timeout = 1000,
+    Workers = [Worker || {_WorkerName, Worker} <- ecpool:workers(PoolName)],
+    DoPerWorker =
+        fun(Worker) ->
+            case ecpool_worker:client(Worker) of
+                {ok, Conn} ->
+                    erlang:is_process_alive(Conn) andalso
+                        ecpool_worker:exec(Worker, fun ?MODULE:do_get_status/1, Timeout);
+                Error ->
+                    Error
+            end
+        end,
+    {ok, Results} =
+        try
+            {ok, emqx_utils:pmap(DoPerWorker, Workers, Timeout)}
+        catch
+            exit:timeout ->
+                {error, timeout}
+        end,
+    sum_worker_results(Results, State).
 
-do_get_status(Conn) ->
-    case eredis:q(Conn, ["PING"]) of
-        {ok, _} -> true;
-        _ -> false
-    end.
+do_cluster_status_check(Pool, State) ->
+    Pongs = eredis_cluster:qa(Pool, [<<"PING">>]),
+    sum_worker_results(Pongs, State).
 
-status_result(_Status = true) -> connected;
-status_result(_Status = false) -> connecting.
+do_get_status(Conn) ->
+    eredis:q(Conn, ["PING"]).
+
+sum_worker_results([], _State) ->
+    connected;
+sum_worker_results([{error, <<"NOAUTH Authentication required.">>} = Error | _Rest], State) ->
+    ?tp(emqx_redis_auth_required_error, #{}),
+    %% This requires user action to fix so we set the status to disconnected
```

Comment:
Maybe we could mark the resource as `unhealthy_target` since it requires operator intervention.

Reply:
I will change to unhealthy_target

Reply:
Fixed